Feature/150398706/receive jwt on signup

fob413 opened the pull request:
What does this PR do?

User receives JWT on successful sign up

Description of Task to be completed?

Users once authenticated via the signup api should be returned a JWT to be required on subsequent call to all other endpoints that require authentication.

How should this be manually tested?

Signup a new user and check the local storage for the jwt token

What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/150398706

Commit:
feat(signup): receive jwt on signup

- users receive jwt on successful signup

[Delivers #150398706]

Commit:
feat(signup): receive jwt on signup

- users receive jwt on successful signup

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- when user successfully signups, he receives a jwt token

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- when user successfully signups, he receives a jwt token

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- user receives jwt on successful signup
- fixed bug that hindered signout button to show

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- user receives jwt on successful signup
- fixed bug that hindered signout button to show

[Delivers #150398706]

Commit:
Merge branch 'feature/150398706/receive-jwt-on-signup' of https://github.com/andela/isildur-cfh into feature/150398706/receive-jwt-on-signup

VanPaitin commented on app/controllers/users.js:
```diff
@@ -82,37 +88,111 @@ exports.checkAvatar = function (req, res) {
+//   } else {
+//     return res.redirect('/#!/signup?error=incomplete');
+//   }
+// };
```
What is this code still doing? If it is not doing anything, you can feel free to remove it. There is no point using ghost code in our project

VanPaitin commented on config/routes.js:
```diff
@@ -23,9 +23,12 @@ module.exports = (app, passport, auth) => {
 
   // Setting up the users api
-  app.post('/users', users.create);
+  // app.post('/users', users.create);
```
Feel free to remove this line

VanPaitin commented on npm-debug.log:
```diff
@@ -0,0 +1,45 @@
+23 error Or if that isn't available, you can get their info via:
+23 error     npm owner ls cardsforhumanity
+23 error There is likely additional logging output above.
+24 verbose exit [ 1, true ]
```
This file should not be commited to version control. Add it to the gitignore

VanPaitin reviewed (changes requested):
Nice work here! Try to fix that long running houndci check and then attend to the ghost codes then I can merge

Commit:
feat(signup): jwt on signup

- removed ghost codes from config/routes.js and  app/controllers/users.js
- git ignored npm-debug.log
- created .eslintignore with the relavant things to be ingnored

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound errors

[Delivers #150398706]

Commit:
feat(signup): jwt on sign up

- fix hound errors

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound errors
- included rule to turn off imort extensions

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error
- added rule to switch off dangling underscore in eslint file

[Delivers #150398706]

Commit:
feat(signup): jwt on sign up

- fix hound errors

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error
- changed underscore rule in eslintrc

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error
- changed underscore rule in eslintrc

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- fix hound error
- changed underscore rule in eslintrc

[Delivers #150398706]

Commit:
feat(signup): jwt on signup

- removed duplication of hashing password

[Delivers #150398706]
